scipy/integrate/_tanhsinh.py

```diff
@@ -915,17 +925,17 @@ def _nsum_iv(f, a, b, step, args, log, maxterms, atol, rtol):
     return f, a, b, step, valid_abstep, args, log, maxterms_int, atol, rtol
 
 
-def _nsum(f, a, b, step=1, args=(), log=False, maxterms=int(2**20), atol=None,
-          rtol=None):
-    r"""Evaluate a convergent sum.
+def nsum(f, a, b, *, step=1, args=(), log=False, maxterms=int(2**20), tolerances=None):
+    r"""Evaluate a convergent, monotonically decreasing finite or infinite series.
```

Comment:
```suggestion
    r"""Evaluate a convergent finite or infinite series.
```
Maybe do this and move the note about "more general forms of series" to the bottom of the extended summary for greater visibility? This could still be quite useful for other kinds of series if the user can do a little bit of manipulation.